modules/identities/scripts/get.ts

```diff
@@ -0,0 +1,58 @@
+import { ScriptContext } from "../module.gen.ts";
+import { IdentityData, IdentityProviderInfo } from "../utils/types.ts";
+
+export interface Request {
+	userToken: string;
+    info: IdentityProviderInfo;
+}
+
+export interface Response {
+    data: {
+        uniqueData: IdentityData;
+        additionalData: IdentityData;
+    } | null;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+	await ctx.modules.rateLimit.throttle({
+        key: req.userToken,
+        period: 10,
+        requests: 10,
+        type: "user",
+    });
+
+    // Ensure the user token is valid and get the user ID
+    const { userId } = await ctx.modules.users.authenticateToken({ userToken: req.userToken } );
+
+    // Get identity data
+    const identity = await ctx.db.userIdentities.findFirst({
+        where: {
+            userId,
+            identityType: req.info.identityType,
+            identityId: req.info.identityId,
+        },
+        select: {
+            uniqueData: true,
+            additionalData: true
+        }
+    });
+
+    // Type checking to make typescript happy
+    const data = identity ?? null;
+    if (!data) {
+        return { data: null };
+    }
+
+    const { uniqueData, additionalData } = data;
```

Comment:
is this because of our ajv validation issue or something else? ideally we don't have to write ugly code like this.

this might already be fixed on main with the zod migration.

Reply:
It's because `uniqueData` and `additionalData` are JSON fields in the psql database.

The null reassignment shouldn't be necessary, that was an accident.

However, it doesn't guarantee there's a matching row, and the JSON type doesn't guarantee it's a non-nullable object.

This essentially just says invalid/missing entries have no data.

Reply:
am i missing something: if there's no matching row, identities will be null. it's not going to return an object with undefined properties. our db constraint has `not null` on jsonb so all of this code is redundant.

Reply:
It sadly is not. The DB constraint says that if there is a row, there is a valid JSON value in that column. `null` is a valid JSON value, along with `"hello"`, `42`, etc.

Prisma ensures that the value returned will be JSON, but not whether or not it's an object or not.

Reply:
to clarify: there is no case where we intentionally set anything that's not an object to the additional data, right? if that's a case, let's write an assertion using the deno assert lib (which throws an internal error, not runtimeerror) & crash instead of failing silentl.

Reply:
otherwise, some downstream script will likely break from unintended state afaik.

Reply:
Good point. Will do!